client/components/auto-direction/index.jsx

```diff
@@ -171,6 +171,9 @@ const setChildDirection = ( child ) => {
 	if ( child && child.props.children ) {
 		let innerChildDirection = null;
 		const children = React.Children.map( child.props.children, innerChild => {
+			if ( ! innerChild ) {
```

Comment:
if you have 
```
<AutoDirection>
  <SomeComponent />
  { false && <OptionalComponent />
</AutoDirection>
```
`AutoDirection` gets two children, `SomeComponent` and `undefined`. The check skips undefined children.